Core/Src/main.c

```diff
@@ -82,7 +96,24 @@ int main(void)
      * 7. Full regression test as much as possible. If don't know what
      * regression test is, read up on it.
      */
-
+	HAL_StatusTypeDef ret;
+	uint8_t hum_val[3]; //humidity raw value
+	uint8_t temp_val[3]; //temperature raw value
+	typedef int32_t BME280_S32_t;
+	typedef uint32_t BME280_U32_t;
+	BME280_S32_t t_fine, adc_T, adc_H;
+	uint8_t spi_buff[5]; //SPI buffer not for measurements
+	uint8_t hum_osrs = 0b00000001; //enable humidity with x1 oversampling

```

Comment:
I would avoid using "magic numbers". In the header file there could be a list of `#define` with all the constant values. For example: 
```
#define ENABLE_HUMIDITY_X1 0x01
#define TURN_FILTER_OFF 0x00
```